pkg/gcmanager/scanAll_IPPool.go

```diff
@@ -120,142 +118,220 @@ func (s *SpiderGC) executeScanAll(ctx context.Context) {
-						if nil != err {
-							wrappedLog.Error(err.Error())
-						}
-						// no matter whether succeed to clean up IPPool IP and SpiderEndpoint, just continue to the next poolIP
+					} else {
+						scanAllLogger.Sugar().Errorf("failed to get pod from kubernetes, error '%v'", err)
 						continue
 					}
-
-					scanAllLogger.Sugar().Errorf("failed to get pod from kubernetes, error '%v'", err)
-					continue
 				}
 
-				// check pod status phase with its yaml
-				podEntry, err := s.buildPodEntry(nil, podYaml, false)
-				if nil != err {
-					scanAllLogger.Sugar().Errorf("failed to build podEntry in scanAll, error: %v", err)
+				if podYaml != nil {
+					flagStaticIPPod = podmanager.IsStaticIPPod(s.gcConfig.EnableStatefulSet, s.gcConfig.EnableKubevirtStaticIP, podYaml)
+				} else {
+					scanAllLogger.Sugar().Errorf("podYaml is nil for pod %s/%s", podNS, podName)
 					continue
 				}
 
-				// case: The pod in IPPool's ip-allocationDetail is also exist in k8s, but the pod is in 'Terminating|Succeeded|Failed' status phase
-				if podEntry != nil {
-					if time.Now().UTC().After(podEntry.TracingStopTime) {
-						wrappedLog := scanAllLogger.With(zap.String("gc-reason", "pod is out of time"))
-						err = s.releaseSingleIPAndRemoveWEPFinalizer(logutils.IntoContext(ctx, wrappedLog), pool.Name, poolIP, poolIPAllocation)
-						if nil != err {
-							wrappedLog.Error(err.Error())
-							continue
+				switch {
+				case podYaml.Status.Phase == corev1.PodSucceeded || podYaml.Status.Phase == corev1.PodFailed:
+					scanAllLogger.Sugar().Infof("The Pod %s/%s status %s is Succeeded or Failed and the IP address needs to be reclaimed", podNS, podName, podYaml.Status.Phase)
+					flagPodStatusShouldGCIP = true
+				case podYaml.DeletionTimestamp != nil:
+					podTracingGracefulTime := (time.Duration(*podYaml.DeletionGracePeriodSeconds) + time.Duration(s.gcConfig.AdditionalGraceDelay)) * time.Second
+					podTracingStopTime := podYaml.DeletionTimestamp.Time.Add(podTracingGracefulTime)
+					if time.Now().UTC().After(podTracingStopTime) {
+						scanAllLogger.Sugar().Infof("the graceful deletion period of pod '%s/%s' is over, try to release the IP address.", podNS, podName)
+						flagPodStatusShouldGCIP = true
+					} else {
+						wrappedLog := scanAllLogger.With(zap.String("gc-reason", "The graceful deletion period of kubernetes Pod has not yet ended"))
+						if len(podYaml.Status.PodIPs) != 0 {
+							wrappedLog.Sugar().Infof("The current Kubernetes Pod %s/%s IPs is not empty, the IP address should not be released, ignore it.", podNS, podName)
+							flagPodStatusShouldGCIP = false
+						} else {
+							wrappedLog.Sugar().Infof("The current Kubernetes Pod %s/%s IPs is empty, try to release the IP address in the IPPool.", podNS, podName)
+							flagPodStatusShouldGCIP = true
+							flagTracePodEntry = true
 						}
+					}
+				default:
+					if len(podYaml.Status.PodIPs) != 0 {
+						flagPodStatusShouldGCIP = false
```

Comment:
// comment ： pod is running， pod has been assigned IP address

Reply:
```
if len(podYaml.Status.PodIPs) != 0 {
						// pod is running, pod has been assigned IP address
						wrappedLog.Sugar().Debugf("pod %s/%s has been assigned IP address %v, ignore handle IP %s", podNS, podName, podYaml.Status.PodIPs, poolIP)
						flagPodStatusShouldGCIP = false
```